VideoLocker/src/main/java/org/edx/mobile/module/serverapi/IRequestImpl.java

```diff
@@ -0,0 +1,78 @@
+package org.edx.mobile.module.serverapi;
+
+import android.os.Bundle;
+
+/**
+ * Created by rohan on 2/6/15.
+ */
+class IRequestImpl implements IRequest {
```

Comment:
Yes, we are using apache HTTP library. But, we want HTTP component to be detached from our code so that it can be easily modified/replaced with any other alternative, if required.  
For example, android recommends <code>HttpURLConnection</code> over apache HTTP. So, we may want to use it in future.